### ISSUE TITLE
Rendering code replaces `@xyz` with github link in URLs, breaking link targets

### ISSUE THREAD

**Issue by joshtriplett** [rust-lang/rust-project-goals#204]
If you write something like `[this document](https://hackmd.org/@xyz/document)`, the rendering code will replace `@xyz` with a github link, breaking the URL in the hackmd link.

This replacement needs to be markdown-aware, and shouldn't happen in (at least) code blocks and link targets. If that's not easily possible, then we should disable the username-replacement logic until it can be fixed.

https://github.com/rust-lang/rust-project-goals/pull/203 was a PR that resulted from having to work around this breakage.

**Comment by nikomatsakis** [comment]
Yeah, it's currently just some simple regexs. It might be worth breaking out a real markdown parser.

**Comment by nikomatsakis** [comment]
It *seems* like comrak can be easily made to do what is needed...

https://github.com/kivikakk/comrak?tab=readme-ov-file#usage
